--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/API/JSStringRef.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/API/JSStringRef.h	2012-09-11 19:27:36.000000000 -0500
@@ -37,7 +37,7 @@
 extern "C" {
 #endif
 
-#if !defined(WIN32) && !defined(_WIN32) && !defined(__WINSCW__) \
+#if !defined(WIN32) && !defined(_WIN32) && !defined(__CYGWIN__) \
     && !((defined(__CC_ARM) || defined(__ARMCC__)) && !defined(__linux__)) /* RVCT */
 /*!
 @typedef JSChar
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/icu/unicode/platform.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/icu/unicode/platform.h	2012-09-11 19:27:36.000000000 -0500
@@ -210,7 +210,11 @@ typedef unsigned int uint32_t;
 /*===========================================================================*/
 
 #define U_HAVE_WCHAR_H      1
+#if defined(__CYGWIN__)
+#define U_SIZEOF_WCHAR_T    2
+#else
 #define U_SIZEOF_WCHAR_T    4
+#endif
 
 #define U_HAVE_WCSCPY       1
 
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/icu/unicode/umachine.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/icu/unicode/umachine.h	2012-09-11 19:27:36.000000000 -0500
@@ -298,7 +298,7 @@ typedef int8_t UBool;
 #       if (U_SIZEOF_WCHAR_T==4)
 #           define U_WCHAR_IS_UTF32
 #       endif
-#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__CYGWIN__)
 #       define U_WCHAR_IS_UTF16    
 #   endif
 #endif
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/JITStubs.cpp	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/JITStubs.cpp	2012-09-11 19:28:57.369532000 -0500
@@ -68,7 +68,7 @@ using namespace std;
 
 namespace JSC {
 
-#if OS(DARWIN) || (OS(WINDOWS) && CPU(X86))
+#if OS(DARWIN) || ((OS(WINDOWS) || OS(CYGWIN)) && CPU(X86))
 #define SYMBOL_STRING(name) "_" #name
 #else
 #define SYMBOL_STRING(name) #name
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/JITStubs.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/JITStubs.h	2012-09-11 19:27:36.000000000 -0500
@@ -116,10 +116,10 @@ namespace JSC {
         ReturnAddressPtr* returnAddressSlot() { return reinterpret_cast<ReturnAddressPtr*>(this) - 1; }
     };
 #elif CPU(X86)
-#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #pragma pack(push)
 #pragma pack(4)
-#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
     struct JITStackFrame {
         void* reserved; // Unused
         JITStubArg args[6];
@@ -143,9 +143,9 @@ namespace JSC {
         // When JIT code makes a call, it pushes its return address just below the rest of the stack.
         ReturnAddressPtr* returnAddressSlot() { return reinterpret_cast<ReturnAddressPtr*>(this) - 1; }
     };
-#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#if COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #pragma pack(pop)
-#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC))
+#endif // COMPILER(MSVC) || (OS(WINDOWS) && COMPILER(GCC)) || OS(CYGWIN)
 #elif CPU(ARM_THUMB2)
     struct JITStackFrame {
         JITStubArg reserved; // Unused
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/ThunkGenerators.cpp	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/jit/ThunkGenerators.cpp	2012-09-12 03:23:13.066467800 -0500
@@ -111,7 +111,7 @@ MacroAssemblerCodeRef sqrtThunkGenerator
     return jit.finalize(*globalData, globalData->jitStubs->ctiNativeCall());
 }
 
-#if OS(DARWIN) || (OS(WINDOWS) && CPU(X86))
+#if OS(DARWIN) || ((OS(WINDOWS) || OS(CYGWIN)) && CPU(X86))
 #define SYMBOL_STRING(name) "_" #name
 #else
 #define SYMBOL_STRING(name) #name
@@ -121,7 +121,7 @@ MacroAssemblerCodeRef sqrtThunkGenerator
 #define SYMBOL_STRING_RELOCATION(name) #name "@plt"
 #elif OS(DARWIN) || (CPU(X86_64) && COMPILER(MINGW) && !GCC_VERSION_AT_LEAST(4, 5, 0))
 #define SYMBOL_STRING_RELOCATION(name) "_" #name
-#elif CPU(X86) && COMPILER(MINGW)
+#elif CPU(X86) && (COMPILER(MINGW) || OS(CYGWIN))
 #define SYMBOL_STRING_RELOCATION(name) "@" #name "@4"
 #else
 #define SYMBOL_STRING_RELOCATION(name) #name
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/PageAllocationAligned.cpp	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/PageAllocationAligned.cpp	2012-09-12 13:52:00.307677100 -0500
@@ -50,7 +50,7 @@ PageAllocationAligned PageAllocationAlig
     vm_map(current_task(), &address, size, alignmentMask, flags, MEMORY_OBJECT_NULL, 0, FALSE, protection, PROT_READ | PROT_WRITE | PROT_EXEC, VM_INHERIT_DEFAULT);
     return PageAllocationAligned(reinterpret_cast<void*>(address), size);
 #else
-    size_t alignmentDelta = alignment - pageSize();
+    size_t alignmentDelta = abs(alignment - pageSize());
 
     // Resererve with suffcient additional VM to correctly align.
     size_t reservationSize = size + alignmentDelta;
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/Platform.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/Platform.h	2012-09-11 19:27:36.000000000 -0500
@@ -312,6 +312,11 @@
 #define WTF_OS_AIX 1
 #endif
 
+/* OS(CYGWIN) - Cygwin */
+#ifdef __CYGWIN__
+#define WTF_OS_CYGWIN 1
+#endif
+
 /* OS(DARWIN) - Any Darwin-based OS, including Mac OS X and iPhone OS */
 #ifdef __APPLE__
 #define WTF_OS_DARWIN 1
@@ -389,6 +394,7 @@
 /* OS(UNIX) - Any Unix-like system */
 #if   OS(AIX)              \
     || OS(ANDROID)          \
+    || OS(CYGWIN)           \
     || OS(DARWIN)           \
     || OS(FREEBSD)          \
     || OS(LINUX)            \
@@ -640,6 +646,7 @@
 
 #if !OS(WINDOWS) && !OS(SOLARIS) && !OS(QNX) \
     && !OS(RVCT) \
+    && !OS(CYGWIN) \
     && !OS(ANDROID)
 #define HAVE_TM_GMTOFF 1
 #define HAVE_TM_ZONE 1
--- origsrc/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/ThreadIdentifierDataPthreads.cpp	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/JavaScriptCore/wtf/ThreadIdentifierDataPthreads.cpp	2012-09-11 20:30:25.692643000 -0500
@@ -45,7 +45,7 @@
 
 namespace WTF {
 
-pthread_key_t ThreadIdentifierData::m_key = PTHREAD_KEYS_MAX;
+pthread_key_t ThreadIdentifierData::m_key;
 
 void clearPthreadHandleForIdentifier(ThreadIdentifier);
 
--- origsrc/webkit-efl-svn-r98964/Source/WebCore/icu/unicode/platform.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/WebCore/icu/unicode/platform.h	2012-09-11 19:27:36.000000000 -0500
@@ -210,7 +210,11 @@ typedef unsigned int uint32_t;
 /*===========================================================================*/
 
 #define U_HAVE_WCHAR_H      1
+#if defined(__CYGWIN__)
+#define U_SIZEOF_WCHAR_T    2
+#else
 #define U_SIZEOF_WCHAR_T    4
+#endif
 
 #define U_HAVE_WCSCPY       1
 
--- origsrc/webkit-efl-svn-r98964/Source/WebCore/icu/unicode/umachine.h	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/WebCore/icu/unicode/umachine.h	2012-09-11 19:27:36.000000000 -0500
@@ -298,7 +298,7 @@ typedef int8_t UBool;
 #       if (U_SIZEOF_WCHAR_T==4)
 #           define U_WCHAR_IS_UTF32
 #       endif
-#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64)
+#   elif defined(WIN32) || defined(_WIN32) || defined(WIN64) || defined(_WIN64) || defined(__CYGWIN__)
 #       define U_WCHAR_IS_UTF16    
 #   endif
 #endif
--- origsrc/webkit-efl-svn-r98964/Source/WebCore/plugins/PluginDatabase.cpp	2011-11-01 10:39:01.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/WebCore/plugins/PluginDatabase.cpp	2012-09-11 19:27:36.000000000 -0500
@@ -445,7 +445,9 @@ void PluginDatabase::getPluginPathsInDir
     // FIXME: This should be a case insensitive set.
     HashSet<String> uniqueFilenames;
 
-#if defined(XP_UNIX)
+#if defined(__CYGWIN__)
+    String fileNameFilter("*.dll");
+#elif defined(XP_UNIX)
     String fileNameFilter("*.so");
 #else
     String fileNameFilter("");
--- origsrc/webkit-efl-svn-r98964/Source/WebKit/CMakeLists.txt	2011-11-01 10:39:02.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/WebKit/CMakeLists.txt	2012-09-12 04:02:30.714300700 -0500
@@ -81,4 +81,4 @@ IF (VERSION_SCRIPT)
 ENDIF ()
 
 SET_TARGET_PROPERTIES(${WebKit_LIBRARY_NAME} PROPERTIES VERSION ${PROJECT_VERSION} SOVERSION ${PROJECT_VERSION_MAJOR})
-INSTALL(TARGETS ${WebKit_LIBRARY_NAME} DESTINATION lib)
+INSTALL(TARGETS ${WebKit_LIBRARY_NAME} LIBRARY DESTINATION lib RUNTIME DESTINATION bin ARCHIVE DESTINATION lib)
--- origsrc/webkit-efl-svn-r98964/Source/cmake/WebKitHelpers.cmake	2011-11-01 10:39:02.000000000 -0500
+++ src/webkit-efl-svn-r98964/Source/cmake/WebKitHelpers.cmake	2012-09-11 19:30:15.159639400 -0500
@@ -14,7 +14,7 @@ MACRO(WEBKIT_SET_EXTRA_COMPILER_FLAGS _t
         SET(OLD_COMPILE_FLAGS "${OLD_COMPILE_FLAGS} -fno-tree-sra")
     ENDIF ()
 
-    IF (NOT SHARED_CORE)
+    IF (NOT SHARED_CORE AND NOT CYGWIN)
         SET(OLD_COMPILE_FLAGS "-fPIC -fvisibility=hidden ${OLD_COMPILE_FLAGS}")
     ENDIF ()
 
